build/downloader/src/cli.ts

```diff
@@ -30,42 +30,43 @@ export const initProject = async (packageLocation: string, common: CommonArgs) =
 
   const packageJson = await fse.readJson(packageLocation)
   for (const toolName of Object.keys(toolsList)) {
-    let forceDownload = false
-    let toolVersion = packageJson[toolName]?.version || packageJson[toolName]
-    const devBranch = packageJson[toolName]?.devBranch
-
-    if (!toolVersion) {
+    if (!packageJson[toolName]) {
       logger.info(`Version missing for tool ${toolName} in package.json`)
       continue
     }
 
-    const releases = await getReleasedFiles(toolName, common.platform)
-    const devRelease = devBranch && releases.find(x => x.version.endsWith(devBranch))
+    let forceDownload = false
+    let binaryInfo: ProcessedRelease | undefined
+    const { version, devBranch } = packageJson[toolName]
+
+    const releases = await getBinaries(toolName, common.platform, devBranch)
+    const devRelease = devBranch && releases.find(x => x.version.endsWith(sanitizeBranch(devBranch)))
 
     if (devBranch && devRelease && !yn(process.env.IGNORE_DEV_BRANCH)) {
       logger.info(`Using the binary of branch "${devBranch}"`)
-      toolVersion = devBranch
+      binaryInfo = devRelease
 
       const existingFile = await getFileMetadata(devRelease, common.appData)
 
       if (existingFile?.fileId !== devRelease.fileId) {
         forceDownload = true
         logger.info('A new binary was published for that branch, re-downloading it...')
       }
+    } else {
+      binaryInfo = releases.find(x => x.version.endsWith(version))
     }
 
-    const configVersion = releases.find(x => x.version.endsWith(toolVersion))
-    if (!configVersion) {
+    if (!binaryInfo) {
       logger.info("Version not found on the tool's release page")
       continue
     }
 
-    const location = path.resolve(common.appData, 'tools', toolName, configVersion?.fileName)
+    const location = path.resolve(common.appData, 'tools', toolName, binaryInfo.fileName)
```

Comment:
It would be nice if there was a way to clear this "cache" of binaries automatically. I think we should remove the older versions when downloading a new one. What do you think?

Reply:
Right now you can type `yarn bpd clean` to remove everything. Auto cleanup would be nice, could be done in another PR since it's the current behavior